Block/Adminhtml/Group/Edit/Form.php

```diff
@@ -0,0 +1,236 @@
+        );
+
+        if ($customerGroup->getId() == 0 && $customerGroup->getCode()) {
+            $name->setDisabled(true);
+        }
+
+        $fieldset->addField(
+            'tax_class_id',
+            'select',
+            [
+                'name' => 'tax_class',
+                'label' => __('Tax Class'),
+                'title' => __('Tax Class'),
+                'class' => 'required-entry',
+                'required' => true,
+                'values' => $this->_taxCustomer->toOptionArray(),
+            ]
+        );
+        $fieldset->addField(
+            'enable_balance_payments',
+            'checkbox',
+            [
+                'name' => 'enable_balance_payments',
+                'label' => __('Enable Balance payments'),
+                'title' => __('Enable Balance payments'),
+                'class' => '',
+                'value' => 1,
+                'checked' => $this->checkCustomerGroup($groupId),
+            ]
+        );
+
+        $fieldset->addField(
+            'customer_group_excluded_website_ids',
+            'multiselect',
+            [
+                'name' => 'customer_group_excluded_websites',
+                'label' => __('Excluded Website(s)'),
+                'title' => __('Excluded Website(s)'),
+                'required' => false,
+                'can_be_empty' => true,
+                'values' => $this->systemStore->getWebsiteValuesForForm(),
+                'note' => __('Select websites you want to exclude from this customer group.')
+            ]
+        );
+
+        if ($customerGroup->getId() !== null) {
+            // If edit add id
+            $form->addField('id', 'hidden', ['name' => 'id', 'value' => $customerGroup->getId()]);
+        }
+
+        if ($this->_backendSession->getCustomerGroupData()) {
+            $form->addValues($this->_backendSession->getCustomerGroupData());
+            $this->_backendSession->setCustomerGroupData(null);
+        } else {
+            // TODO: need to figure out how the DATA can work with forms
+            $form->addValues(
+                [
+                    'id' => $customerGroup->getId(),
+                    'customer_group_code' => $customerGroup->getCode(),
+                    'tax_class_id' => $defaultCustomerTaxClass,
+                    'customer_group_excluded_website_ids' => $customerGroupExcludedWebsites
+                ]
+            );
+        }
+
+        $form->setUseContainer(true);
+        $form->setId('edit_form');
+        $form->setAction($this->getUrl('customer/*/save'));
+        $form->setMethod('post');
+        $this->setForm($form);
+    }
+
+    /**
+     * CheckCustomerGroup
+     *
+     * @param $groupId
+     * @return bool
+     * @throws NoSuchEntityException
+     */
+    public function checkCustomerGroup($groupId)
+    {
+        $arrCustomerGroup = [];
+        $allowedCustomerGroups = $this->config->getAllowedCustomerGroups();
+        if (!empty($allowedCustomerGroups)) {
```

Comment:
No need for this condition, if `$allowedCustomerGroups` is empty `foreach` statement will not execute